src/main/java/org/javarosa/core/model/TriggerableDag.java

```diff
@@ -392,41 +392,16 @@ private static void throwCyclesInDagException(Collection<QuickTriggerable> trigg
      * Get all of the elements which will need to be evaluated (in order) when
      * the triggerable is fired.
      */
-    private static Set<QuickTriggerable> getDependantTriggerables(FormInstance mainInstance, EvaluationContext ec, QuickTriggerable triggerable, Map<TreeReference, Set<QuickTriggerable>> triggerIndex) {
-        Set<QuickTriggerable> allDependantTriggerables = new LinkedHashSet<>();
-        Set<TreeReference> targets = new HashSet<>();
-        for (TreeReference target : triggerable.getTargets()) {
-
-            targets.add(target);
-
-            // For certain types of triggerables, the update will affect
-            // not only the target, but also the children of the target.
-            // In that case, we want to add all of those nodes
-            // to the list of updated elements as well.
-            if (triggerable.isCascadingToChildren()) {
-                targets.addAll(getChildrenOfReference(mainInstance, ec, target));
-            }
-        }
-
-        // Now go through each of these updated nodes (generally
-        // just 1 for a normal calculation,
-        // multiple nodes if there's a relevance cascade.
-        for (TreeReference target : targets) {
-            // Check our index to see if that target is a Trigger
-            // for other conditions
-            // IE: if they are an element of a different calculation
-            // or relevancy calc
-
-            // We can't make this reference generic before now or
-            // we'll lose the target information,
-            // so we'll be more inclusive than needed and see if any
-            // of our triggers are keyed on the predicate-less path
-            // of this ref
-            Set<QuickTriggerable> dependantTriggerables = triggerIndex.get(target.hasPredicates() ? target.removePredicates() : target);
-            if (dependantTriggerables != null)
-                allDependantTriggerables.addAll(dependantTriggerables);
+    private static Set<QuickTriggerable> getDependantTriggerables(QuickTriggerable triggerable, Map<TreeReference, Set<QuickTriggerable>> triggerIndex) {
+        Set<QuickTriggerable> result = new HashSet<>();
+        for (TreeReference targetRef : triggerable.getTargets()) {
+            Set<QuickTriggerable> children = triggerIndex.getOrDefault(targetRef, emptySet());
```

Comment:
I need to come back to 86e2745 and think through it.

Reply:
I've added a commit to deal with this.

Also, the commits belonging to 86e2745 could be split into another PR if you want.